helga_jeopardy.py

```diff
@@ -315,34 +669,125 @@ def jeopardy(client, channel, nick, message, cmd, args,
+
+                clue_active = sel_category + '.' + clue_idx + '.active'
+                mongo_db.update({
+                    'game_active': True,
+                    'channel': channel,
+                }, {
+                    '$set': {
+                        'control': nick,
+                        clue_active: False,
+                        nick: new_score,
+                    }
+                })
+
+                if check_remaining_clues(client, channel) is False:
+                    return
+
+                reactor.callLater(1, show_board, client, channel)
+
+                if last_control == nick:
+                    return random.choice(game_correct_responses_maintain).format(nick)
+
+                else:
+                    return random.choice(game_correct_responses_take).format(nick)
+
+
+            else:
+                mongo_db.update({
+                    'active': True,
+                    'channel': channel,
+                }, {
+                    '$set': {
+                        'active': False,
+                        'answered_by': nick,
+                        'timestamp': datetime.datetime.utcnow(),
+                    }
+                })
+
+                return random.choice(correct_responses).format(nick)
 
         if partial > 0:
             return u"{}, can you be more specific?".format(nick)
 
-        # wrong answer, ignore
+        # wrong answer, ignore for random q, but deduct if game
+
+        else:
+            if current_game:
+                if nick not in current_game['players']:
+                    client.msg(channel, "You did not join this game. Please wait for the next game to join")
+                    return
+                score = current_game[nick]
+                new_score = score - question['value']
+                mongo_db.update({
+                    'game_active': True,
+                    'channel': channel,
+                }, {
+                    '$set': {
+                        nick: new_score,
+                    }
+                })
+                return
+            
         return
 
     if question and not args:
         logger.debug('no answer provided :/')
         return
 
     if not question and args:
-        logger.debug('no active question :/')
-        return
-
-    question_text = quest_func(client, channel)
+        if args[0] != "game" and not args[0].startswith('cat'):
+            logger.debug('no active question :/')
+            return
+
+    if args[0].startswith('cat'):
+        if len(args) != 2:
+            client.msg(channel, "You must provide a category and a value")
+            return
+        if not current_game:
+            client.msg(channel, "Game not started.  to join: !j game join   to start: !j game start")
+            return
+
+        if nick not in current_game['players']:
+            client.msg(channel, "Game in progress. You must wait for the next game to join")
```

Comment:
Are you intentionally using different error messages for different situations where the user did not join the game? If not, I would recommend creating a constant to store the "you are not a part of this game" error and use that instead of having them sprinkled throughout the code.

Reply:
Not in this case, I've now replaced this one with a list of responses for its 3 occurrences . Most others are specific to the function they appear in, although some may be repeated once.